### PR TITLE
Optimizely late experiment activation

### PR DESCRIPTION
**Pull request by davidbarna**
In optimizely, experiments activation can be triggered by user interaction.
`<AbTestOptimizelyXExperiment>` was not prepared and worked only for immediately activated experiments.
Now it works for both cases.

### CODE REVIEW

File: components/abtesting/optimizelyXExperiment/src/optimizely-x.js [hunk 1]
```diff
@@ -19,12 +24,48 @@ const waitUntil = (truthyFn, callback, delay = 100, interval = 100) => {
...
+ * @param {Object} sdk OptimizelySdk
+ */
+const registerExprimentsEvents = sdk => {
+  if (!wasExprimentsListenerAdded) {
+    sdk.push({
+      type: 'addListener',
+      filter: {type: 'lifecycle', name: 'campaignDecided'},
+      handler: exprimentsActivationHandler
+    })
+    wasExprimentsListenerAdded = true
```
Comment (kikoruiz) [inline components/abtesting/optimizelyXExperiment/src/optimizely-x.js]: Typo here, maybe you wanted to mean `wasExperimentsListenerAdded `.

File: components/abtesting/optimizelyXExperiment/src/optimizely-x.js [hunk 2]
```diff
@@ -19,12 +24,48 @@ const waitUntil = (truthyFn, callback, delay = 100, interval = 100) => {
...
+  }
+}
+
+/**
+ * Handler for Optimizely events
+ * @param {Object} event
+ */
+const exprimentsActivationHandler = event => {
```
Comment (kikoruiz) [inline components/abtesting/optimizelyXExperiment/src/optimizely-x.js]: Same here ;)

**Review by kikoruiz — DISMISSED** [review]
Nice job!

File: components/abtesting/optimizelyXExperiment/src/index.js [hunk 3]
```diff
@@ -3,14 +3,24 @@ import PropTypes from 'prop-types'
...
+import OptimizelyX from './optimizely-x'
 
 class AbTestOptimizelyXExperiment extends Component {
-  state = {variation: undefined}
+  state = {variation: null}
 
   componentDidMount() {
-    this._experiment = createExperimentUseCase(this.props.experimentId)
-    this._experiment.execute().then(variation => this.setState({variation}))
+    this._activationHandler = variation => this.setState({variation})
```
Comment (midudev) [inline components/abtesting/optimizelyXExperiment/src/index.js:12]: Why are you creating the _activationHandler class property inside the componentDidMount instead extracting it directly to the class?

Comment (davidbarna) [reply components/abtesting/optimizelyXExperiment/src/index.js:12]: To be as lazy as possible.
I create this handler only when I need it.

File: components/abtesting/optimizelyXExperiment/src/optimizely-x.js (hunk 1 above)
Comment (davidbarna) [reply components/abtesting/optimizelyXExperiment/src/optimizely-x.js]: 🍋 🍊 exprimentation XD

**Review by devictoribero — APPROVED** [review]
Took me a while to read everything!
Looks good! Good job! 👍 🥇

**Review by kikoruiz — APPROVED** [review]

**Pull request merged**